src/main/java/com/customerApi/api/services/CustomerService.java

```diff
@@ -0,0 +1,150 @@
+package com.customerApi.api.services;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.hibernate.exception.JDBCConnectionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.customerApi.api.exception.CustomerNotFoundException;
+import com.customerApi.api.modals.Customer;
+import com.customerApi.api.repositories.CustomerRepository;
+
+@Service
+public class CustomerService {
+
+	private static final Logger logger = LoggerFactory.getLogger(CustomerService.class);
+
+	@Autowired
+	private CustomerRepository customerRepo;
+
+	public Optional<Integer> addCustomerData(Customer customer) {
+
+		Optional<Integer> customerId = null;
+
+		try {
+
+			Customer cust = customerRepo.save(customer);
+
+			customerId = Optional.of(cust.getId());
+
```

Comment:
Fix this logic here, why you need to pull customerId, if you can return the full customer object.